src/core/control/LatexController.cpp

```diff
@@ -118,9 +118,9 @@ void LatexController::findSelectedTexElement() {
         this->posx = rect.x;
         this->posy = rect.y;
 
-        if (auto* img = dynamic_cast<TexImage*>(this->selectedElem)) {
+        if (auto* img = dynamic_cast<TexImage*>(this->selectedElem); img) {
             this->initialTex = img->getText();
-        } else if (auto* txt = dynamic_cast<Text*>(this->selectedElem)) {
+        } else if (auto* txt = dynamic_cast<Text*>(this->selectedElem); img) {
```

Comment:
```suggestion
        } else if (auto* txt = dynamic_cast<Text*>(this->selectedElem); txt) {
```

Reply:
With certain compilers assignment if-statements produce a warning and it is best practice to ether wrap the assignment in extra brackets (`if ((...))`, source e.g. [llvm](https://llvm.org/docs/CodingStandards.html#treat-compiler-warnings-like-errors)) or (as it's done in other parts in the code) do it with the `;`.
I did the second option, to fit the rest of the code.